crates/driver/src/domain/competition/mod.rs

```diff
@@ -313,6 +392,149 @@ impl Competition {
         Ok(score)
     }
 
+    // Oders already need to be sorted from most relevant to least relevant so that
+    // we allocate balances for the most relevants first.
+    async fn sort_orders(
```

Comment:
This is what causes the weird double `.await` a couple of lines up.
```suggestion
    fn sort_orders(
```